### ISSUE TITLE
FeaFusformer中的代码问题

### ISSUE THREAD

**Issue by PrinceWang01** [Caoxuheng/HIFtool#20]
作者您好，学习了你的HIFtool框架，感觉收获很多！我在学习FeaFusformer代码的时候，有一个地方感觉是不是表述有些问题？

```
# HIFtool-main/Networks/FeafusFormer/Model/spe_down.py
def initialize_SpeDNet(module, msi, hsi, sf):
    # ......省略.......
    for epoch in range(max_epochs):
        trainer.zero_grad()
        pre_msi = module(hsi_1)
        psnr = L1(msi_1, pre_msi)
        psnr.backward()
        trainer.step()
        lrsched.step()

    print('Initialize Spectral Degradation Net Successfully. Epoch:{}\tlr:{:.2e}\tPSNR:{:.2f}'.format(epoch,trainer.state_dict()['param_groups'][0]['lr'],-psnr.detach().cpu().numpy()))
    return module
```
从print这里的描述来看，需要输出PSNR值，但是根据代码的逻辑，这个psnr变量似乎只是L1损失？要想看PSNR的话，需要另外计算一个值出来才对吧？
在spa_down.py里面使用的PSNR损失，所以直接取值就行，但是spe_down里面我感觉是要改一下。

**Comment by Caoxuheng** [comment]
是的，这里输出的是L1，PSNR需要自己写，这是个BUG。



在 2025-08-13 09:27:24，"PrinceWang01" ***@***.***> 写道：

PrinceWang01 created an issue (Caoxuheng/HIFtool#20)

作者您好，学习了你的HIFtool框架，感觉收获很多！我在学习FeaFusformer代码的时候，有一个地方感觉是不是表述有些问题？

# HIFtool-main/Networks/FeafusFormer/Model/spe_down.py
def initialize_SpeDNet(module, msi, hsi, sf):
    # ......省略.......
    for epoch in range(max_epochs):
        trainer.zero_grad()
        pre_msi = module(hsi_1)
        psnr = L1(msi_1, pre_msi)
        psnr.backward()
        trainer.step()
        lrsched.step()

    print('Initialize Spectral Degradation Net Successfully. Epoch:{}\tlr:{:.2e}\tPSNR:{:.2f}'.format(epoch,trainer.state_dict()['param_groups'][0]['lr'],-psnr.detach().cpu().numpy()))
    return module


从print这里的描述来看，需要输出PSNR值，但是根据代码的逻辑，这个psnr变量似乎只是L1损失？要想看PSNR的话，需要另外计算一个值出来才对吧？
在spa_down.py里面使用的PSNR损失，所以直接取值就行，但是spe_down里面我感觉是要改一下。
